### ISSUE TITLE
SequenceChunk.js redundant access to chunks.

### ISSUE THREAD

**Issue by enuggetry** [GMOD/jbrowse#641]
I noticed an inefficiency in SequenceChunks. This is the store class that manages refseq access.  Requests that fall in the same chunk, result in the pulling the same chunk from the server multiple times.  From what I've seen, it can pull the same chunk like 7+ times.
Should cache the last chuck to resolve it.

**Comment by cmdcolin** [comment]
That can become a concern sometimes but most of the time it seems to fetch the chunks from the browsers cache too

I always check the network tab on the chrome bar to check this


<img width="1407" alt="cache" src="https://cloud.githubusercontent.com/assets/6511937/10553828/914b090a-7423-11e5-9f28-e894768a36d7.png">

Here the black box shows the things that are pulled from cache and the red arrow shows the total network traffic. The things that are pulled from cache don't increase the network traffic

Might be interesting if the chunks were cached without relying on this browser cache though! I am not aware of what might be blocking that

**Comment by enuggetry** [comment]
Ah, right.  Maybe that's why it was left that way.

**Comment by nathandunn** [comment]
I keep wondering (reading through #658) if the client-code could . . like @cmdcolin  suggested, just internally query / store cache and only make a request if its not there.

**Comment by cmdcolin** [comment]
Might be relevant to other track types as well, e.g. the REST API will get a request for each chunk too https://github.com/GMOD/jbrowse/issues/395

**Comment by einhverfr** [comment]
Whether it grabs from the browser cache depends on the headers sent.  In my case this is actually sent via a web service and for this reason the default headers say not to cache.

I think this works for content served in a certain way, but it has some limits as a more general tool.  For example, when browsing a contig of several megabases from a web service that says not to cache, this becomes a problem.  And since this is stored in a relational db, doing chunking would still require a lot of page/cache management on the db side.

So I would like to contribute another store which effectively memoizes this call.  Would be happy to contribute it.

**Comment by einhverfr** [comment]
After my previous pull request was found to have some gaps in functionality, I tried setting caching headers.  I tested in firefox, chrome, and internet explorer.  What I found suggests this issue really does need to be resolved.

In Chrome, it is true, the first request hits the physical resource and all subsequent requests hit cache.  This is the only browser I tested where this behavior was observed.  All other browsers suffered from a specific issue here.

Firefox and Internet Explorer would send multiple requests concurrenly.  Several requests would be made before the cache would be updated and ould be hit.  One the first request was complete, then subsequent requests would hit cache but it was not uncommon to see 5 or 10 requests for the same refseq in these cases.  I really have to support Internet Explorer here, and some of the reference sequences are 3 megabases long, retrieved from a text field in a relational database.

The problem I am running into is there seems to be no provision for caching in these stores, and so every chunk requests a new request.
